crates/aptos-rest-client/src/lib.rs

```diff
@@ -556,15 +556,15 @@ impl Client {
                 }
                 WaitForTransactionResult::Pending(state) => {
                     if expiration_timestamp_secs <= state.timestamp_usecs / 1_000_000 {
-                        return Err(anyhow!("Transaction expired, it is guaranteed it will not be committed on chain.").into());
+                        return Err(anyhow!("Transaction expired (pending), it is guaranteed it will not be committed on chain.").into());
                     }
                     chain_timestamp_usecs = Some(state.timestamp_usecs);
                 }
                 WaitForTransactionResult::NotFound(error) => {
                     if let RestError::Api(aptos_error_response) = error {
                         if let Some(state) = aptos_error_response.state {
                             if expiration_timestamp_secs <= state.timestamp_usecs / 1_000_000 {
-                                return Err(anyhow!("Transaction expired, it is guaranteed it will not be committed on chain.").into());
+                                return Err(anyhow!("Transaction expired (not found), it is guaranteed it will not be committed on chain.").into());
                             }
                             chain_timestamp_usecs = Some(state.timestamp_usecs);
                         }
```

Comment:
let's change
debug!(
"Cannot yet find transaction in mempool on {:?}, continuing to wait.",

to warn!